packages/expo-modules-core/android/src/main/java/expo/modules/adapters/react/apploader/RNHeadlessAppLoader.kt

```diff
@@ -22,16 +24,35 @@ class RNHeadlessAppLoader @DoNotStrip constructor(private val context: Context)
     }
 
     if (context.applicationContext is ReactApplication) {
-      val reactInstanceManager = (context.applicationContext as ReactApplication).reactNativeHost.reactInstanceManager
```

Comment:
@chrfalch previously we were casting `context.applicationContext`, but it was changed to casting `context`, which I assumed was not intentional?